### DISCUSSION TITLE
Not re-enabling automerge after Renovate PR is kicked out of merge queue

### DISCUSSION THREAD

**Discussion by tommy** [renovatebot/renovate#25153] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

We are using Renovate (and very happy with it!) alongside GitHub-native merge queues and the renovate-approve bot. I've noticed recently some Renovate PRs remaining stuck open and requiring a manual merge. 

My understanding of what's happening is:

1) a renovate PR is created/approved/added to merge queue (great)
2) is kicked out of the merge queue due to a conflict with the base branch (very common for projects using an NPM lockfile, when renovate creates PRs for multiple dependencies in a short time\)
3) renovate updates the PR and force-pushes, so the PR is green again (great)
4) but, it doesn’t get put back into the queue, so the PR sits open.

Below is a screenshot of the activity on such a PR.

Is there any way to configure Renovate to automatically re-enable "Merge When Ready" after rebasing a PR like this?

![CleanShot 2023-10-11 at 18 10 22@2x](https://github.com/renovatebot/renovate/assets/27465/7a6e4382-e72a-420d-9297-8723a5857d2b)

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment]
Should it retry forever? There may be "permanent" reasons why it fails, so renovate would consume your Actions time forever by continuously putting it back into the merge queue.

If not retrying forever, we'd need a way of counting how many times it's been tried, but I'm not sure that GitHub's API returns that data

**Reply by tommy** [reply]
I guess I would just expect it to re-enable "merge when ready" at the same time that it rebases the PR. It won't end up on the queue unless the checks pass on the PR again, which I think would mean it avoids an infinite retry loop. Maybe that doesn't work for users who don't have PR checks?

**Reply by rarkins** [reply]
I think sometimes people have different checks on PRs versus the main branch so you could end up with an infinite loop of it passing PR checks and failing merge queue ones.

Also there may be cases where the branch fails the merge queue but _doesn't_ need rebasing because it's not conflicted, so we need to consider that too.

In short, there's a lot of permutations here and it's not as simple as "put it back on the merge queue"

**Reply by tommy** [reply]
That makes sense. Thanks for the help!

**Reply by rarkins** [reply]
BTW I'm not saying "no", just that "it's complicated". I think we-enabling automerge on rebase hopefully is ok, for example
